comp/core/autodiscovery/proto/proto.go

```diff
@@ -0,0 +1,96 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package proto provides autodiscovery proto util functions
+package proto
+
+import (
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo/core"
+)
+
+// ProtobufConfigFromAutodiscoveryConfig converts an autodiscovery config to a protobuf config
+func ProtobufConfigFromAutodiscoveryConfig(config *integration.Config) *core.Config {
+	instances := [][]byte{}
+
+	for _, instance := range config.Instances {
+		instances = append(instances, []byte(instance))
+	}
+
+	advancedAdIdentifiers := make([]*core.AdvancedADIdentifier, 0, len(config.AdvancedADIdentifiers))
+	for _, advancedAdIdentifier := range config.AdvancedADIdentifiers {
+		advancedAdIdentifiers = append(advancedAdIdentifiers, &core.AdvancedADIdentifier{
+			KubeService: &core.KubeNamespacedName{
+				Name:      advancedAdIdentifier.KubeService.Name,
+				Namespace: advancedAdIdentifier.KubeService.Namespace,
+			},
+			KubeEndpoints: &core.KubeNamespacedName{
+				Name:      advancedAdIdentifier.KubeEndpoints.Name,
+				Namespace: advancedAdIdentifier.KubeEndpoints.Namespace,
+			},
+		})
+	}
+
+	return &core.Config{
+		Name:                    config.Name,
+		Instances:               instances,
+		InitConfig:              config.InitConfig,
+		MetricConfig:            config.MetricConfig,
+		LogsConfig:              config.LogsConfig,
+		AdIdentifiers:           config.ADIdentifiers,
+		AdvancedAdIdentifiers:   advancedAdIdentifiers,
+		Provider:                config.Provider,
+		ServiceId:               config.ServiceID,
+		TaggerEntity:            config.TaggerEntity,
+		ClusterCheck:            config.ClusterCheck,
+		NodeName:                config.NodeName,
+		Source:                  config.Source,
+		IgnoreAutodiscoveryTags: config.IgnoreAutodiscoveryTags,
+		MetricsExcluded:         config.MetricsExcluded,
+		LogsExcluded:            config.LogsExcluded,
+	}
+}
+
+// AutodiscoveryConfigFromProtobufConfig converts a protobuf config to an autodiscovery config
+func AutodiscoveryConfigFromProtobufConfig(config *core.Config) integration.Config {
```

Comment:
`config` is a pointer, so it might be `nil`. 

Maybe it is better to make a check here. 

I am not sure about the intended behaviour in this case. If the intended behaviour is having a panic, it will panic anyways with `nil pointer dereference` error message. If the intended behaviour is to return an empty `integration.Config` then it should be handled. 

In both cases, I think it can be good to check if `config` is nil.

Reply:
I went with an empty config to avoid the panic 👍